advisor-backend/routes/schedule.js

```diff
@@ -39,12 +39,78 @@ const calculateInDegrees = (
   }
 };
 
+const checkDesiredYearAndBefore = (
+  scheduleObject,
+  desiredYear,
+  postReqsSet
+) => {
+  let validMoveFlag = true;
+  for (const year in scheduleObject) {
+    // only check desiredYear and before
+    if (parseInt(year) <= desiredYear) {
+      const coursesSet = scheduleObject[year];
+      postReqsSet.forEach((postReq) => {
+        if (coursesSet.has(postReq)) {
+          // invalid schedule
+          validMoveFlag = false;
+        }
+      });
+    }
+  }
+  return validMoveFlag;
+};
+
+const checkDesiredYearAndAfter = (scheduleObject, desiredYear, preReqsSet) => {
+  let validMoveFlag = true;
+  for (const year in scheduleObject) {
+    // only check desiredYear and after
+    if (parseInt(year) >= desiredYear) {
+      const coursesSet = scheduleObject[year];
+      preReqsSet.forEach((preReq) => {
+        if (coursesSet.has(preReq)) {
+          // invalid schedule
+          validMoveFlag = false;
+        }
+      });
+    }
+  }
+  return validMoveFlag;
+};
```

Comment:
The only difference between these two is the direction/check that happens on line 67; you could combine these and pass in either a boolean like `forwardLooking` or something, or an enum value. 
```
const isValidSchedule = (scheduleObject, year, requiredClasses, forwardLooking = true) => {
  let validMoveFlag = true;
  for (const year in scheduleObject) {
    // only check the target year and before/after 
    if (forwardLooking ? parseInt(year) >= desiredYear :  parseInt(year) <= desiredYear) {
      const coursesSet = scheduleObject[year];
      requiredClasses.forEach((req) => {
        if (coursesSet.has(req)) {
          // invalid schedule
          validMoveFlag = false;
        }
      });
    }
  }
  return validMoveFlag;
};
```